lib/plugins/physics.js

```diff
@@ -51,25 +54,42 @@ function inject (bot, { physicsEnabled }) {
 
   // This function should be executed each tick (every 0.05 seconds)
   // How it works: https://gafferongames.com/post/fix_your_timestep/
+
+  // WARNING: THIS IS NOT ACCURATE ON WINDOWS (15.6 Timer Resolution)
+  // use WSL or switch to Linux
+  // see: https://discord.com/channels/413438066984747026/519952494768685086/901948718255833158
   let timeAccumulator = 0
   function doPhysics () {
     const now = performance.now()
-    const deltaSeconds = (now - lastPhysicsFrameTime) / 1000
+    let deltaSeconds = (now - lastPhysicsFrameTime) / 1000
     lastPhysicsFrameTime = now
 
+    if (deltaSeconds >= PHYSICS_CATCHUP_TICKS) deltaSeconds = PHYSICS_CATCHUP_TICKS // limiter (limit catchups)
```

Comment:
```js
deltaSeconds = Math.min(deltaSeconds, PHYSICS_CATCHUP_TICKS)
```

Reply:
That is not the right place to limit the number of catchup ticks, you should add a counter to the while loop bellow and have the PHYSICS_CATCHUP_TICKS be in tick units. Otherwise you'll make the timeAccumulator have a wrong value and desync.

Reply:
I am confused on where to put the counter and how to limit it properly.
```js
  let timeAccumulator = 0
  let catchupTicks = 0
  function doPhysics () {
    const now = performance.now()
    let deltaSeconds = (now - lastPhysicsFrameTime) / 1000
    lastPhysicsFrameTime = now

    timeAccumulator += deltaSeconds
    catchupTicks = 0
    while (timeAccumulator >= PHYSICS_TIMESTEP) {
      tickPhysics(now)
      timeAccumulator -= PHYSICS_TIMESTEP
      catchupTicks++
      if (catchupTicks >= PHYSICS_CATCHUP_TICKS) timeAccumulator -= PHYSICS_TIMESTEP * PHYSICS_CATCHUP_TICKS
    }
  }
  ```
  Please clarify for me.

Reply:
Almost good, but you shouldn't remove time in the last if, just break and let the next tick continue catching up.

If you remove time you'll skip ticks, causing desync.

Reply:
Thanks for clarification